### DISCUSSION TITLE
Wall mote set values not working

### DISCUSSION THREAD

**Discussion by Robbie1983** [zwave-js/node-red-contrib-zwave-js#161] [Q&A]
Hi everyone, 

I am slowly moving from wired (self made mqtt based domitica) to zwave based commercial switches as the wife wants a more stylish looking switch assembly. However, I am struggling a little with the configuration of the different switches. For example a aeotec wallmote. 

The wallmote is succesfully added and working, sending notifications and also the values are accessible for reading.
However, I am unable to set the configuration parameters both through CCapi or ValueApi.

For example the  touch sound should be adjustable trough the "configuration" parameter=1.

![image](https://user-images.githubusercontent.com/95997962/145704433-db17a218-fa32-4213-947e-a4d6bcb1ca21.png)

however sending:

let Message = {
 payload: {
 mode: "CCAPI",
 node: 5,
 cc: "Configuration",
 method: "set",
 params: [1,0,1]
 }
}
return Message

Will not set the touch sound to disable. Also the other configuration parameters seem not to be editable. 
Any help is appreciated......

**Comment by marcus-j-davies** [comment]
Morning @Robbie1983,

Are the values changing when modifying them within the UI tab?

**Comment by marcus-j-davies** [comment]
@Robbie1983 

Just noticed its a battery operated device, config params will get applied when the device awakens on a schedule, or you wake the device up - using a combination of button presses of sorts (user manual).

commands will be queued for sleeping devices, and will get sent when they wake up.
the device sending a notification, is not the same as it waking up to receive config updates.

from the manual of wall mote.

![image](https://user-images.githubusercontent.com/55892693/145705595-16075baa-cdc5-463e-a145-16b28f91ccb6.png)

Battery operated devices, also wake up on a schedule - if they support Wake UP scheduling - you can change its interval on the UI, but it will need to be awake to receive 😆

**Reply by Robbie1983** [reply]
Thank you, that did the trick indeed, Still getting used to these controllers. Perhaps, I could also bother you with something else that I cannot figure out. I do not get a notification of the sliding action of the wallmote. It seems more people are struggling with this in different implementation. Do you have a fix for this?

Kind regards,

Rob

**Comment by marcus-j-davies** [comment]
Hi @Robbie1983 - Sure no problem.

first here is the config file for the wallmote (it might not be the exact wallmote version - which one do you have?)

https://devices.zwave-js.io/?jumpTo=0x0086:0x0002:0x0081:0.0

Parameter 3 mentions slide function

**Reply by Robbie1983** [reply]
Dear Marcus, 

I have the Wallmote quad, model zw130-c, the sliding action is enabled and the wallmote gives a signal when sliding. 
Paramater 3 is set to enabled. 

Kind regards,

Rob

**Reply by marcus-j-davies** [reply]
Ok,

I have an idea, Aeotec may be using Groups for dimming functions.
![image](https://user-images.githubusercontent.com/55892693/145707543-9a9a4563-7ca4-464f-a36b-ed1ebf967970.png)

With this in mind, can you do the following (your device will need to be awake during all this).

1 - Open up Association Management UI for the wallmote (option under the node info)
2 - Choose Group 3 (dimming notifications for Button 1)
3 - Add a new association for this group  set it to (Node ID  - 1, Endpoint -  0)

what we are doing here, is targeting the controller (Node 1) as the receiver for dimming - allowing you to react

Device Info Entry for quad (also lists the groups)
https://devices.zwave-js.io/?jumpTo=0x0086:0x0002:0x0082:0.0

**Reply by Robbie1983** [reply]
I tried this with the remote awake however, It indicates it is not able to add the association.

**Reply by marcus-j-davies** [reply]
what error did you get?
devices don't stay awake - see further comment below on how to keep them awake.

**Reply by marcus-j-davies** [reply]
You could also try.

Node ID 1 (as before) but target Endpoint 1 as the association target.
have just tried this on a different Aeotec device (door sensor) and it worked.

**Comment by marcus-j-davies** [comment]
If you want to keep you device awake.

```javascript
let Message = {
    payload: {
        mode: "ControllerAPI",
        method: "keepNodeAwake",
        params: [<NodeID>, true]
    }
}
return Message
```

 - wake up the device after (so it stays awake)
 - remember to reverse it (So it is allowed to go back to sleep) - wake it up again after reversing it (Evan though it will already be awake)

**Reply by marcus-j-davies** [reply]
Battery devices are a little annoying when configuring them at first - but once all set, you don't have to worry about waking them up, as they will have been configured the way you want.

**Reply by Robbie1983** [reply]
It worked, it is really nice to have people to help you, my previous setup was completely without commercial things. Which also leaves it up to yourself to figure everything out. Your help is really appreciated!

**Reply by marcus-j-davies** [reply]
Nice!

so you get the slide changes now?

**Reply by Robbie1983** [reply]
I am figuring out which notification to monitor to be honest

**Reply by Robbie1983** [reply]
As currently no notifications are showing up when sliding

**Reply by marcus-j-davies** [reply]
Cool.

on the basis our association for Button 1 worked
you can do the same for all 4 

Group 3 - Button 1 slide
Group 5 - Button 2 slide
Group 7 - Button 3 slide
Group 9 - Button 3 slide

The lifeline (Group 1) - will take care of the basics (ON/OFF) - which is usually setup up automatically.

Edit!
> As currently no notifications are showing up when sliding

Oh try endpoint 1 for the association Node ID 1, Endpoint 1
this works with a test device I have.

**Reply by marcus-j-davies** [reply]
Example (button 1 slide)

![image](https://user-images.githubusercontent.com/55892693/145709225-83680af9-c407-4e24-9e74-6cecde5bc472.png)

**Reply by Robbie1983** [reply]
Unfortunately still no notification from node 1

**Reply by marcus-j-davies** [reply]
Could you show me what you have for group 3?

**Reply by marcus-j-davies** [reply]
we might also need to set Param 4 to 3

According to manual

> The WallMote Quad also supports sliding control, when the parameter 4 [1 byte] has been set to 3, you can slide up or down on the button area to control a dimmer device by sliding up or down. Swiping up once will dim up a dimmer device by a certain percentage, while swiping down once will dim down a dimmer device by a certain percentage.

**Reply by marcus-j-davies** [reply]
In theory.

 - Setting groups 3,5,7,9 to Node ID 1, Endpoint 1
 - Setting Param 3 to 1
 - Setting Param 4 to 3
 
 All of which can be done in the UI - Should give you slide control
 
 Attach a debug node to the controller

**Reply by Robbie1983** [reply]
I will try it later today and let you know if it worked

**Reply by Robbie1983** [reply]
Works, thank you!

**Reply by marcus-j-davies** [reply]
Cool!

shout if you need help with anything 👍
